otel-loadbalancer/main.go

```diff
@@ -130,7 +129,9 @@ func newSharder(ctx context.Context) (*sharder.Sharder, *lbdiscovery.Manager, er
 		sharder.SetTargets(targets)
 		sharder.Reshard()
 	})
-	sharder.SetCollectors(collectors)
+	k8sClient.Watch(ctx, cfg.LabelSelector, func(collectors []string) {
+		sharder.SetCollectors(collectors)
```

Comment:
sharder.SetCollectors is not safe for concurrent usage. We need to protect the collectors with a mutex in the sharder before landing this PR.